README.md
# singleuser
🕵️‍♂️ Jupyter-singleuser docker image prebuild, as base for Naas 

# build

`docker build -t jupyternaas/singleuser:latest .`

# dev

`docker run --rm -it -p 8888:8888 -e JUPYTER_ENABLE_LAB=yes jupyternaas/singleuser:latest`


CHANGELOG.md
## 2.12.0 (2022-02-18)

### Feat

- Build ARM version as well

### Fix

- Bump jupyterlab 3.0.16

## 2.11.19 (2021-08-17)

### Fix

- **jupyter/minimal-notebook-version**: Fix to 4d9c9bd9ced0

## 2.11.18 (2021-08-17)

### Fix

- **Dependency**: Bump nbdime

## 2.11.17 (2021-08-16)

### Refactor

- disable the jupyterlab tour

## 2.11.16 (2021-04-29)

### Fix

- jupyterlab version

## 2.11.15 (2021-04-29)

### Fix

- remove useless packages

## 2.11.14 (2021-04-08)

### Fix

- remove delete of work folder

## 2.11.13 (2021-04-08)

### Fix

- remove duplicated module for now

## 2.11.12 (2021-04-07)

### Fix

- add back python-language

## 2.11.11 (2021-04-07)

### Fix

- remove useless files

## 2.11.10 (2021-04-07)

### Fix

- machine user

## 2.11.9 (2021-04-07)

### Fix

- autocompletion module

## 2.11.8 (2021-04-07)

### Fix

- ci

## 2.11.7 (2021-04-06)

### Fix

- get-chagelog-diff version

## 2.11.6 (2021-04-06)

### Fix

- add changelog output in slack

## 2.11.5 (2021-04-06)

### Fix

- get-changelog lastChanges

## 2.11.4 (2021-04-02)

### Fix

- --no-cache-dir

## 2.11.3 (2021-04-01)

### Fix

- remove useless docker command

## 2.11.2 (2021-04-01)

### Fix

- remove useless folder

## 2.11.1 (2021-03-27)

### Fix

- add missing copy step

## 2.11.0 (2021-03-27)

### Feat

- ci add dependabot

## 2.10.0 (2021-03-27)

### Feat

- add cheagelog output changes
- switch to requirements file

## 2.9.2 (2021-03-27)

### Fix

- get last fix from jupyterlab 3

## 2.9.1 (2021-03-27)

### Fix

- update mathplotlib to latest version

## 2.9.0 (2021-03-25)

### Feat

- add xeu python

## 2.8.2 (2021-03-23)

### Fix

- ci

## 2.8.1 (2021-03-23)

### Fix

- ci

## 2.8.0 (2021-03-23)

### Feat

- add all lib for puperteer

## 2.7.0 (2021-03-23)

### Feat

- add missing lib for pupertter

## 2.6.9 (2021-03-18)

### Fix

- tag commit

## 2.6.8 (2021-03-18)

### Fix

- commit message

## 2.6.7 (2021-03-18)

### Fix

- slack notif

## 2.6.6 (2021-03-18)

### Fix

- typo in file comited

## 2.6.5 (2021-03-18)

### Fix

- remove useless step

## 2.6.4 (2021-03-18)

### Fix

- ci add back missing step

## 2.6.3 (2021-03-18)

### Fix

- build ci

## 2.6.2 (2021-03-17)

### Fix

- duplicate step

## 2.6.1 (2021-03-17)

### Fix

- add ci step to update config file in naas repo

## 2.6.0 (2021-03-17)

### Fix

- remove buggy changelog
- jupyter-server-proxy version
- comment archive for now
- slack message

### Feat

- add slack notif for build

## 2.6.0b12 (2021-03-12)

### Fix

- remove bug changelog
- add missing libs

## 2.6.0b11 (2021-03-11)

### Fix

- changelog

### Feat

- add package for lab v3

## 2.6.0b10 (2021-03-11)

### Fix

- eror in changelog
- flake8

## 2.6.0b9 (2021-03-11)

### Fix

- version for jhub

## 2.6.0b8 (2021-03-11)

### Fix

- remove useless flag

## 2.6.0b7 (2021-03-11)

### Fix

- remove elyra

## 2.6.0b6 (2021-03-11)

### Fix

- dep errors

## 2.6.0b5 (2021-03-11)

### Fix

- remove useless nass config

## 2.6.0b4 (2021-03-11)

### Fix

- remove changelog
- use fixed version
- pip install

## 2.6.0b3 (2021-03-11)

### Fix

- remove changelog
- jupyter-resource-usage dep

## 2.6.0b2 (2021-03-10)

### Fix

- changelog remove it
- add missing spreadsheet

## 2.6.0b1 (2021-03-10)

### Fix

- remove changelog broken
- version link

### Feat

- add python-language-server for auto complete

## 2.6.0b0 (2021-03-10)

### Feat

- use v3

## 2.5.1 (2021-02-16)

### Fix

- remove naas config from here

## 2.5.0 (2021-02-16)

### Feat

- add onboarding file path

### Fix

- css sidebar

## 2.4.1 (2021-02-16)

### Fix

- add missing dependency for quick open

## 2.4.0 (2021-02-16)

### Feat

- add css custom module

## 2.3.4 (2021-02-12)

### Fix

- use naas folder, instead of jupyter for naas assets

## 2.3.3 (2021-02-04)

### Fix

- remove naas from singleuser

## 2.3.2 (2021-02-04)

### Fix

- remove useless dir

## 2.3.1 (2021-02-04)

### Fix

- error when missing env var

## 2.3.0 (2021-01-25)

### Feat

- add wkhtmltopdf to build

## 2.2.0 (2021-01-25)

### Feat

- add spreadsheet editor

## 2.1.4 (2021-01-21)

### Fix

- reactive jupyter-launcher-shortcuts

## 2.1.2b22 (2021-01-19)

## 2.1.3 (2021-01-19)

### Fix

- overrides

## 2.1.2 (2021-01-12)

### Fix

- remove buggy changeling

## 2.1.2b21 (2021-01-12)

### Fix

- **ci**: error in step who decide branch to call naas

## 2.1.2b20 (2021-01-11)

### Fix

- ci build jupyterlab

## 2.1.2b19 (2021-01-11)

### Fix

- build prod mod again

## 2.1.2b18 (2021-01-11)

### Fix

- ci release for dev

## 2.1.2b17 (2021-01-11)

### Fix

- deploy ci script

## 2.1.2b16 (2021-01-11)

## 2.1.2b15 (2021-01-07)

### Fix

- try old version without plugin shortcut

## 2.1.2b14 (2021-01-07)

### Fix

- remove shortcut for now

## 2.1.2b13 (2021-01-07)

### Fix

- try jupyter version 3

## 2.1.2b12 (2021-01-07)

### Fix

- better nodejs limit

## 2.1.2b11 (2021-01-07)

### Fix

- try to don't minimize

## 2.1.2b10 (2021-01-07)

### Fix

- split run to fin error easily

## 2.1.2b9 (2021-01-07)

### Fix

- remove useless pip flag

## 2.1.2b8 (2021-01-07)

### Fix

- nodejs build size

## 2.1.2b7 (2021-01-07)

### Fix

- nodejs build

## 2.1.2b6 (2021-01-07)

### Fix

- version number jup client

## 2.1.2b5 (2021-01-07)

### Fix

- version jupyter

## 2.1.2b4 (2021-01-05)

### Fix

- jupyterlab build

## 2.1.2b3 (2021-01-04)

### Fix

- ovverride config

## 2.1.2b2 (2020-12-10)

### Feat

- Export KERNEL_ variables
- use LauncherShortcuts

### Fix

- name manager shortcut
- make overrides work again
- overrides

## 2.1.0 (2020-11-09)

## 1.3.10 (2020-11-09)

### Fix

- event to build

## 1.3.9 (2020-11-09)

### Fix

- update readme

## 1.3.8 (2020-10-27)

### Fix

- logo typo in css

## 1.3.7 (2020-10-27)

### Fix

- remove overrides for now

## 1.3.6 (2020-10-27)

### Fix

- css
- remove useless vars

## 1.3.5 (2020-10-27)

### Fix

- path

## 1.3.4 (2020-10-27)

### Fix

- css append

## 1.3.3 (2020-10-27)

### Fix


jupyter_server_config.py
# Copyright (c) Naas Team.
# Distributed under the terms of the Modified BSD License.

from jupyter_core.paths import jupyter_data_dir
import subprocess
import os
import errno
import stat

c = get_config()
c.ServerApp.ip = '0.0.0.0'
c.ServerApp.port = 8888
c.ServerApp.default_url = '/lab'

# c.LabBuildApp.minimize = False
# c.LabBuildApp.dev_build = False

c.ServerApp.webbrowser_open_new = 0

c.ServerApp.tornado_settings = {
    'headers': {
        'Content-Security-Policy': f"frame-ancestors self {os.environ.get('ALLOWED_IFRAME')}"
    }
}

# c.LauncherShortcuts.shortcuts = {
#     'naas_manager': {
#         'title': 'Naas Manager',
#         'target': '{base_url}naas/',
#         'icon_path': '/etc/jupyter/naas_logo.svg',
#     }
# }

c.ServerProxy.servers = {
    'naas': {
        'launcher_entry': {
            'enabled': True,
            'icon_path': '/etc/naas/naas_logo.svg',
            'title': 'Naas manager',
        },
        'new_browser_tab': False,
        'timeout': 30,
        'command': ["redir", "--lport={port}", "--cport=5000"],
    }
}

# Generate a self-signed certificate
if 'GEN_CERT' in os.environ:
    dir_name = jupyter_data_dir()
    pem_file = os.path.join(dir_name, 'notebook.pem')
    try:
        os.makedirs(dir_name)
    except OSError as exc:  # Python >2.5
        if exc.errno == errno.EEXIST and os.path.isdir(dir_name):
            pass
        else:
            raise

    # Generate an openssl.cnf file to set the distinguished name
    cnf_file = os.path.join(
        os.getenv('CONDA_DIR', '/usr/lib'), 'ssl', 'openssl.cnf')
    if not os.path.isfile(cnf_file):
        with open(cnf_file, 'w') as fh:
            fh.write('''\
[req]
distinguished_name = req_distinguished_name
[req_distinguished_name]
''')

    # Generate a certificate if one doesn't exist on disk
    subprocess.check_call(['openssl', 'req', '-new',
                           '-newkey', 'rsa:2048',
                           '-days', '365',
                           '-nodes', '-x509',
                           '-subj', '/C=XX/ST=XX/L=XX/O=generated/CN=generated',
                           '-keyout', pem_file,
                           '-out', pem_file])
    # Restrict access to the file
    os.chmod(pem_file, stat.S_IRUSR | stat.S_IWUSR)
    c.ServerApp.certfile = pem_file

# Change default umask for all subprocesses of the notebook server if set in
# the environment
if 'NB_UMASK' in os.environ:
    os.umask(int(os.environ['NB_UMASK'], 8))


jupyter_notebook_config.py
# Copyright (c) Naas Team.
# Distributed under the terms of the Modified BSD License.

from jupyter_core.paths import jupyter_data_dir
import subprocess
import os
import errno
import stat

c = get_config()
c.NotebookApp.ip = '0.0.0.0'
c.NotebookApp.port = 8888
c.NotebookApp.default_url = '/lab'

# c.LabBuildApp.minimize = False
# c.LabBuildApp.dev_build = False

c.NotebookApp.webbrowser_open_new = 0

c.NotebookApp.tornado_settings = {
    'headers': {
        'Content-Security-Policy': 'frame-ancestors self ' + os.environ.get('ALLOWED_IFRAME', '')
    }
}

# c.LauncherShortcuts.shortcuts = {
#     'naas_manager': {
#         'title': 'Naas Manager',
#         'target': '{base_url}naas/',
#         'icon_path': '/etc/jupyter/naas_logo.svg',
#     }
# }

c.ServerProxy.servers = {
    'naas': {
        'launcher_entry': {
            'enabled': True,
            'icon_path': '/etc/naas/naas_logo.svg',
            'title': 'Naas manager',
        },
        'new_browser_tab': False,
        'timeout': 30,
        'command': ["redir", "--lport={port}", "--cport=5000"],
    }
}

# Generate a self-signed certificate
if 'GEN_CERT' in os.environ:
    dir_name = jupyter_data_dir()
    pem_file = os.path.join(dir_name, 'notebook.pem')
    try:
        os.makedirs(dir_name)
    except OSError as exc:  # Python >2.5
        if exc.errno == errno.EEXIST and os.path.isdir(dir_name):
            pass
        else:
            raise

    # Generate an openssl.cnf file to set the distinguished name
    cnf_file = os.path.join(
        os.getenv('CONDA_DIR', '/usr/lib'), 'ssl', 'openssl.cnf')
    if not os.path.isfile(cnf_file):
        with open(cnf_file, 'w') as fh:
            fh.write('''\
[req]
distinguished_name = req_distinguished_name
[req_distinguished_name]
''')

    # Generate a certificate if one doesn't exist on disk
    subprocess.check_call(['openssl', 'req', '-new',
                           '-newkey', 'rsa:2048',
                           '-days', '365',
                           '-nodes', '-x509',
                           '-subj', '/C=XX/ST=XX/L=XX/O=generated/CN=generated',
                           '-keyout', pem_file,
                           '-out', pem_file])
    # Restrict access to the file
    os.chmod(pem_file, stat.S_IRUSR | stat.S_IWUSR)
    c.NotebookApp.certfile = pem_file

# Change default umask for all subprocesses of the notebook server if set in
# the environment
if 'NB_UMASK' in os.environ:
    os.umask(int(os.environ['NB_UMASK'], 8))
